PURL-TYPES.rst

```diff
@@ -297,6 +297,29 @@ github
       pkg:github/package-url/purl-spec@244fd47e07d1004
       pkg:github/package-url/purl-spec@244fd47e07d1004#everybody/loves/dogs
 
+github-release
+--------------
+``github-release`` for GitHub releases:
+
+- ``namespace``: The GitHub user or organization. It is not case sensitive and
+  must be lowercased.
+- ``name``: The GitHub repository name. It is not case sensitive and must be
+  lowercased.
+- ``version``: The release version. It is required and is case sensitive.
+- Qualifier ``repository_url``: GitHub server hosting the release (optional).
+  Useful in case a private server is used. If omitted, ``https://github.com``
+  is assumed as the default.
+- Qualifier ``file_name``: Selects a named (case sensitive) asset contained 
+  within the release (optional).
+- Qualifier ``checksum``: Checksum for the release asset (optional). Must be
+  in the form of `lowercase_algorithm:hex_encoded_lowercase_value`.
+
+- Examples::
+
+      pkg:github-release/cli/cli@v2.67.0
+      pkg:github-release/foo/bar@v1.0.0?repository_url=https:%2F%2Ffoobar.ghe.com
+      pkg:github-release/foo/bar@v1.0.0?file_name=bin-linux.tgz&checksum=sha256:deadbeef
```

Comment:
Is there anything that could not be accomplished with the `github` type and a `file_name` qualifier?

Reply:
Concrete example of why this is different from the github type: `pkg:github-release/keepassxc/keepassxc@2.7.10?file_name=KeePassXC-2.7.10-arm64.dmg` from https://github.com/keepassxreboot/keepassxc/releases/tag/2.7.10. The file is not part of the Git repository so it cannot be referenced using `pkg:github`. You'd need to represent it with a `pkg:generic`. I'm not sure what `pkg:github-release` gives beyond `pkg:generic`. GitHub Releases is a place where you could be downloading a package from but isn't a package management system. There's no implied meaning about version numbers or the content or format of the file or dependencies between releases.

Reply:
It seems weird to me for `pkg:github/keepassxreboot/keepassxc@2.7.10` to refer to a tag in a Git repository and `pkg:github/keepassxreboot/keepassxc@2.7.10?file_name=KeePassXC-2.7.10-arm64.dmg` to refer to an arbitrary file that exists alongside that Git repository. I guess it's kind of similar to `pkg:pypi` where the package is made of multiple files that are not necessarily related, except that for `pkg:github` the `file_name` qualifier would change the mechanism for retrieving the package.